pkg/cmd/cli/restore/delete.go

```diff
@@ -17,31 +17,196 @@ limitations under the License.
+	# delete restores named "restore-1" and "restore-2"
+	ark restore delete restore-1 restore-2
 
-			err = arkClient.ArkV1().Restores(f.Namespace()).Delete(name, nil)
-			cmd.CheckError(err)
+	# delete all restores triggered by schedule "schedule-1"
+	ark restore delete --selector ark-schedule=schedule-1
+	
+	# delete all restores
+	ark restore delete --all`,
+
+		Args: cobra.ExactArgs(1),
+		Run: func(c *cobra.Command, args []string) {
+			cmd.CheckError(opts.Complete(f, args))
+			cmd.CheckError(opts.Validate(c, f, args))
+			cmd.CheckError(opts.Run())
 
-			fmt.Printf("Restore %q deleted\n", name)
 		},
 	}
-
+	opts.BindFlags(c.Flags())
 	return c
 }
+
+// DeleteOptions contains parameters used for deleting a restore.
+type DeleteOptions struct {
+	Names     []string
+	All       bool
+	Selector  flag.LabelSelector
+	Confirm   bool
+	client    clientset.Interface
+	namespace string
+}
+
+// Complete fills in the correct values for all the options.
+func (opts *DeleteOptions) Complete(f client.Factory, args []string) error {
+	opts.namespace = f.Namespace()
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+	opts.client = client
+	opts.Names = args
+	return nil
+
+}
+
+// Validate validates the fields of the DeleteOptions struct.
+func (opts *DeleteOptions) Validate(c *cobra.Command, f client.Factory, args []string) error {
+	if opts.client == nil {
+		return errors.New("Ark client is not set; unable to proceed")
+	}
+	var (
+		hasNames    = len(opts.Names) > 0
+		hasAll      = opts.All
+		hasSelector = opts.Selector.LabelSelector != nil
+	)
+	if !xor(hasNames, hasAll, hasSelector) {
+		return errors.New("you must specify exactly one of: specific backup name(s), the --all flag, or the --selector flag")
+	}
+
+	return nil
+
+}
+
+// xor returns true if exactly one of the provided values is true,
+// or false otherwise.
+func xor(val bool, vals ...bool) bool {
+	res := val
+
+	for _, v := range vals {
+		if res && v {
+			return false
+		}
+		res = res || v
+	}
+
+	return res
+}
+
+// Run performs the deletion of restore(s).
+func (opts *DeleteOptions) Run() error {
+	if !opts.Confirm && !getConfirmation() {
+		return nil
+	}
+	var (
+		restores []*arkv1api.Restore
+		errs     []error
+	)
+
+	switch {
+	case len(opts.Names) > 0:
+		for _, name := range opts.Names {
+			restore, err := opts.client.ArkV1().Restores(opts.namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				errs = append(errs, errors.WithStack(err))
+				continue
+			}
+			restores = append(restores, restore)
+		}
+	default:
+		selector := labels.Everything().String()
+		if opts.Selector.LabelSelector != nil {
+			selector = opts.Selector.String()
+		}
+		res, err := opts.client.ArkV1().Restores(opts.namespace).List(metav1.ListOptions{
+			LabelSelector: selector,
+		})
+		if err != nil {
+			errs = append(errs, errors.WithStack(err))
+		}
+
+		for i := range res.Items {
+			restores = append(restores, &res.Items[i])
+		}
+
+		if len(restores) == 0 {
```

Comment:
This line til the end of this default switch need to be moved to outside of the switch statement.